support/Environments/expanse_gcc.sh

```diff
@@ -0,0 +1,375 @@
+        echo "Charm++ is already installed"
+    else
+        echo "Installing Charm++..."
+        wget https://github.com/UIUC-PPL/charm/archive/v6.10.2.tar.gz
+        tar xzf v6.10.2.tar.gz
+        mv charm-6.10.2 charm
+        cd $dep_dir/charm
+        ./build charm++ verbs-linux-x86_64-smp --with-production -j4
+        ./build EveryLB verbs-linux-x86_64-smp --with-production -j4
+        ./build ScotchLB verbs-linux-x86_64-smp --with-production -j4
+        cd $dep_dir
+        rm v6.10.2.tar.gz
+        echo "Installed Charm++ into $dep_dir/charm"
+        cat >$dep_dir/modules/charm <<EOF
+#%Module1.0
+prepend-path LIBRARY_PATH "$dep_dir/charm/verbs-linux-x86_64-smp/lib"
+prepend-path LD_LIBRARY_PATH "$dep_dir/charm/verbs-linux-x86_64-smp/lib"
+prepend-path CPATH "$dep_dir/charm/verbs-linux-x86_64-smp/include"
+prepend-path CMAKE_PREFIX_PATH "$dep_dir/charm/verbs-linux-x86_64-smp/"
+prepend-path PATH "$dep_dir/charm/verbs-linux-x86_64-smp/bin"
+setenv CHARM_VERSION 6.10.2
+setenv CHARM_HOME $dep_dir/charm/verbs-linux-x86_64-smp
+setenv CHARM_ROOT $dep_dir/charm/verbs-linux-x86_64-smp
+EOF
+    fi
+    cd $dep_dir
+
+    if [ -f $dep_dir/modules/spectre_gsl ]; then
+        echo "GSL is already set up"
+    else
+        # This hard-coded gsl path is not easily maintainable, so if and when
+        # Caltech HPC assigns an accessible environment variable for the loaded
+        # GSL module, we should use that in favor of the hard-coded path
+        cat >$dep_dir/modules/spectre_gsl <<EOF
+#%Module1.0
+setenv GSL_HOME "/software/gsl/2.4"
+setenv GSL_ROOT "/software/gsl/2.4"
+EOF
+    fi
+    cd $dep_dir
+
+    if [ -f $dep_dir/jemalloc/lib/libjemalloc.a ]; then
+        echo "jemalloc is already set up"
+    else
+        echo "Installing jemalloc..."
+        wget https://github.com/jemalloc/jemalloc/releases/download/5.2.1/jemalloc-5.2.1.tar.bz2
+        tar -xjf jemalloc-5.2.1.tar.bz2
+        mv jemalloc-5.2.1 jemalloc
+        cd $dep_dir/jemalloc
+        ./autogen.sh
+        make
+        cd $dep_dir
+        rm jemalloc-5.2.1.tar.bz2
+        echo "Installed jemalloc into $dep_dir/jemalloc"
+        cat >$dep_dir/modules/jemalloc <<EOF
+#%Module1.0
+prepend-path LIBRARY_PATH "$dep_dir/jemalloc/lib"
+prepend-path LD_LIBRARY_PATH "$dep_dir/jemalloc/lib"
+prepend-path CPATH "$dep_dir/jemalloc/include"
+prepend-path PATH "$dep_dir/jemalloc/bin"
+prepend-path LD_RUN_PATH "$dep_dir/jemalloc/lib"
+setenv JEMALLOC_HOME $dep_dir/jemalloc/
+prepend-path CMAKE_PREFIX_PATH "$dep_dir/jemalloc/"
+EOF
+    fi
+
+    cd $start_dir
+
+    spectre_unload_sys_modules
+    module unload scotch
+    module unload binutils
+
+    printf "\n\nIMPORTANT!!!\nIn order to be able to use these modules you\n"
+    echo "must run:"
+    echo "  module use $dep_dir/modules"
+    echo "You will need to do this every time you compile SpECTRE, so you may"
+    echo "want to add it to your ~/.bashrc."
```

Comment:
Do we still want to keep the message about adding this to your .bashrc?  Maybe, since you need it for compiling. But it might cause problems with people having more than one environment. (the way SpEC solves this is that its compile command loads the environment).

Reply:
I'll tweak the message to say "..., so if you want this SpECTRE environment to be your primary environment, you may want to add it to your ~/.bashrc."